packages/bot-web-ui/src/components/dashboard/dashboard-component/load-bot-preview/recent-workspace.tsx

```diff
@@ -0,0 +1,127 @@
+import classnames from 'classnames';
+import React from 'react';
+import { Icon } from '@deriv/components';
+import { timeSince } from '@deriv/bot-skeleton';
+import { save_types } from '@deriv/bot-skeleton/src/constants/save-type';
+import { connect } from 'Stores/connect';
+import RootStore from 'Stores/index';
+import './index.scss';
+
+type TRecentWorkspace = {
+    getRecentFileIcon: (string: string) => void;
+    getSaveType: () => void;
+    previewRecentStrategy: () => void;
+    selected_strategy_id: string;
+    workspace: { [key: string]: string };
+    setFileLoaded: (param: boolean) => void;
+    index: number;
+    has_file_loaded: boolean;
+    recent_strategies: boolean;
+    toggleSaveModal: () => void;
+    onToggleDeleteDialog: (type: string, is_delete_modal_open: boolean) => void;
+    loadFileFromRecent: () => void;
+    setActiveTab: (active_tab: number) => void;
+    toggleStrategies: (param: boolean) => void;
+    is_delete_modal_open: boolean;
+};
+
+const RecentWorkspace = ({
+    getRecentFileIcon,
+    getSaveType,
+    previewRecentStrategy,
+    selected_strategy_id,
+    workspace,
+    index,
+    has_file_loaded,
+    recent_strategies,
+    setFileLoaded,
+    toggleSaveModal,
+    loadFileFromRecent,
+    onToggleDeleteDialog,
+    setActiveTab,
+    toggleStrategies,
+}: TRecentWorkspace) => {
+    const trigger_div_ref = React.useRef<HTMLInputElement | null>(null);
+    React.useEffect(() => {
+        if (index === 0 && has_file_loaded === false) {
+            trigger_div_ref?.current?.click();
+        }
+    }, [has_file_loaded]);
+    return (
+        <>
+            <div
+                className={classnames('load-strategy__recent-item', {
+                    'load-strategy__recent-item__loaded': recent_strategies,
+                    'load-strategy__recent-item--selected': selected_strategy_id === workspace.id,
+                })}
+                key={workspace.id}
+                onClick={
+                    selected_strategy_id === workspace.id
+                        ? undefined
+                        : () => {
+                              previewRecentStrategy(workspace.id);
+                              setFileLoaded(true);
+                          }
```

Comment:
```suggestion
                    () => {
                        if (selected_strategy_id != workspace.id) {
                            previewRecentStrategy(workspace.id);
                            setFileLoaded(true);                        
                        }
                    }
```
Not a fan of assigning `undefined` to `onClick`. You can actually rewrite it this way and it should achieve the same functionality

Reply:
done ✅